test/integration/extension_discovery_integration_test.cc

```diff
@@ -692,5 +707,28 @@ TEST_P(ExtensionDiscoveryIntegrationTest, DestroyDuringInit) {
   ecds_connection_.reset();
 }
 
+// Validate that a listener update should fail if the subscribed extension configuration make filter
+// terminal but the filter position is not at the last position at filter chain.
+TEST_P(ExtensionDiscoveryIntegrationTest, BasicFailTerminalFilterNotFit) {
```

Comment:
The `NotFit` bit here is a bit vague, can you make it clearer from the name what we're testing here?

Reply:
How about BasicFailTerminalFilterNotAtLastInFilterChain?

Reply:
`BasicFailTerminalFilterNotAtEndOfFilterChain` maybe?

Reply:
ok, fixed.